homeassistant/components/camera/bomradarloop.py

```diff
@@ -0,0 +1,381 @@
+    'WaggaWagga':      {'id': '553', 'delta': 600, 'frames': 4},
+    'Warrego':         {'id': '673', 'delta': 600, 'frames': 4},
+    'Warruwi':         {'id': '773', 'delta': 360, 'frames': 6},
+    'Watheroo':        {'id': '793', 'delta': 360, 'frames': 6},
+    'Weipa':           {'id': '783', 'delta': 360, 'frames': 6},
+    'WillisIs':        {'id': '413', 'delta': 600, 'frames': 4},
+    'Wollongong':      {'id': '033', 'delta': 360, 'frames': 6},
+    'Woomera':         {'id': '273', 'delta': 600, 'frames': 4},
+    'Wyndham':         {'id': '073', 'delta': 600, 'frames': 4},
+    'Yarrawonga':      {'id': '493', 'delta': 360, 'frames': 6},
+}
+
+LOCS = sorted(RADARS.keys())
+BADLOC = "Set 'location' to one of: {}".format(', '.join(LOCS))
+LOGGER = logging.getLogger(__name__)
+
+
+def _validate_schema(cfg):
+    msg0 = "Specify either 'id' or 'location', not both"
+    msg1 = "Specify 'id', 'delta' and 'frames' when 'location' is unspecified"
+    if cfg.get('location'):
+        if cfg.get('id'):
+            raise Invalid(msg0)
+    else:
+        if not all([cfg.get('id'), cfg.get('delta'), cfg.get('frames')]):
+            raise Invalid(msg1)
+    return cfg
+
+
+PLATFORM_SCHEMA = All(PLATFORM_SCHEMA.extend({
+    Optional(CONF_DELTA): cv.positive_int,
+    Optional(CONF_OUTFN): cv.string,
+    Optional(CONF_FRAMES): cv.positive_int,
+    Optional(CONF_ID): cv.positive_int,
+    Optional(CONF_LOC): All(In(LOCS), msg=BADLOC),
+    Optional(CONF_NAME): cv.string,
+}), _validate_schema)
+
+
+def _get_url(path):
+    """Return a canonical URL for a suffix path on the BOM website."""
+    _log("Getting URL for path %s", path)
+    return 'http://www.bom.gov.au/{}'.format(path)
+
+
+def _log(*msg):
+    LOGGER.info(*msg)
+
+
+def _log_error(*msg):
+    LOGGER.error(*msg)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up BOM radar-loop camera component."""
+    location = config.get(CONF_LOC)
+    if location:
+        radar_id = RADARS[location]['id']
+        delta = config.get(CONF_DELTA) or RADARS[location]['delta']
+        frames = config.get(CONF_FRAMES) or RADARS[location]['frames']
+    else:
+        radar_id = config.get(CONF_ID)
+        delta = config.get(CONF_DELTA)
+        frames = config.get(CONF_FRAMES)
+        location = "ID {}".format(radar_id)
+    name = config.get(CONF_NAME) or "BOM Radar Loop - {}".format(location)
+    outfn = config.get(CONF_OUTFN)
+    bomradarloop = BOMRadarLoop(
+        hass,
+        location,
+        delta,
+        frames,
+        radar_id,
+        name,
+        outfn
+    )
+    add_devices([bomradarloop])
+
+
+class BOMRadarLoop(Camera):
+    """A camera component producing animated BOM radar-imagery GIFs."""
+
+    def __init__(self, hass, location, delta, frames, radar_id, name, outfn):
+        """Initialize the component."""
+        import PIL.Image
+
+        super().__init__()
+
+        self.hass = hass
```

Comment:
This is done by entity platform, no need to do manually

Reply:
Removed.